### ISSUE TITLE
standardizing github workflows and consoldiating frontend check

### ISSUE THREAD

**Issue by to-sta** [activist-org/activist#431]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

We currently have 3 GitHub workflows (excl. the pr_maintainer_checklist 😄)

1. `pr_backend_formatting_check.yaml`
2. `pr_frontend_formatting_check.yaml`
3. `pr_ts_typecheck.yaml`

<ins>Tasks:</ins>
1. Rename `pr_backend_formatting_check` to `ci_backend`
2. We use different package manager on the frontend side for workflows (yarn, npm), let's standardize that. 

We should also be more specific, here we run black, but what does it actually do? We check the formatting.

```yaml
- name: Run Black
  uses: psf/black@stable
  with:
    options: --check --verbose --exclude "migrations/"
    src: ./backend

- name: Run ruff
  run: ruff ./backend

- name: Run mypy - Static Type Checking
  run: mypy ./backend --config-file ./backend/mypy.ini
```

@andrewtavis 
What you say about consoldating the `pr_ts_typecheck` and `pr_frontend_formatting_check` to `pr_ci_frontend`? 

### Contribution

Happy to answer any question, regarding this issue 😃.

**Comment by andrewtavis** [comment]
Thanks for the issue, @to-sta! I've been thinking about this myself :) I'd be fine with merging the frontend checks, but we should do it in a way that that there's two processes such that if the first one fails the other one still runs. Would be good to keep naming consistent, so maybe for the backend we could name it `pr_ci_backend`? 

Generally yes to all of the above :) Using yarn and npm is another great point that's a long time coming to getting standardized! 🚀

**Comment by to-sta** [comment]
Ok, great! Should have been pr_ci_backend 😄.

> I'd be fine with merging the frontend checks, but we should do it in a way that that there's two processes such that if the first one fails the other one still runs. 

Yeah, that makes sense and implementing it is straight forward, we just need to add `continue-on-error: true`.

**Comment by andrewtavis** [comment]
Fantastic! Looking forward to the changes. I guess we can leave this open for someone to pick up if they have interest and maybe have the goal that's it's done by the next sync?

**Comment by CaioCintra** [comment]
Can I try this one?

**Comment by to-sta** [comment]
@CaioCintra Sure, I have assigned you 👌

**Comment by andrewtavis** [comment]
Closed by #444 and a minor fix in 5e78953 ✅ Appreciate you helping us our with this, @CaioCintra! Hope to work with you again 😊

Thanks also @to-sta for the review! 💪
